records/state/utils.go

```diff
@@ -0,0 +1,63 @@
+package state
```

Comment:
IMO `utils`, `misc` and such packages are a smell. They're the trash bin of programming taxonomy. :-)

These methods belong where the types they act on are. Let's put them in `types.go`. Better still, I'd rename `types.go` to `state.go`